index.js

```diff
@@ -0,0 +1,363 @@
+app.use((req, res, next) => {
+    const auth = req.cookies[AUTH_COOKIE_NAME];
+    if (auth == null) {
+        next();
+        return;
+    }
+
+    const data = jwt.decode(auth);
+    req.user = _.find(that.gammaUsers, ["cid", data.user_name]);
+    next();
+});
+
+var POSSIBLE_STATES = {
+    noVote: "noVote",
+    vote: "vote",
+    result: "result",
+};
+
+const updateGammaCache = () =>
+    new Promise((resolve, reject) => {
+        axios
+            .get(app.get("gammaBaseUrl") + "/users/minified", {
+                headers: { Authorization: "pre-shared " + app.get("apiKey") },
+            })
+            .then(response => {
+                that.gammaUsers = response.data;
+                resolve();
+            })
+            .catch(() => reject());
+    });
+
+updateGammaCache();
+
+const isSignedIn = (req, res, next) => {
+    if (req.user) {
+        next();
+    } else {
+        res.status(401).send(getGammaUri(app));
+    }
+};
+
+const canVote = (req, res, next) => {
+    if (acceptedUsersManager.usersApproved.includes(req.user.id)) {
+        next();
+    } else if (req.user.cid === app.get("adminCid")) {
+        if (!acceptedUsersManager.usersNotApproved.includes(req.user.cid)) {
+            acceptedUsersManager.addToNotApprovedList(req.user.id);
+        }
+        acceptedUsersManager.approveUser(req.user.id);
+        next();
+    } else {
+        const userId = req.user.id;
+
+        if (!acceptedUsersManager.usersNotApproved.includes(userId)) {
+            acceptedUsersManager.addToNotApprovedList(userId);
+        }
+
+        res.status(401).send("You cannot vote yet");
+    }
+};
+
+const isAdmin = (req, res, next) => {
+    if (req.user.cid === app.get("adminCid")) {
+        next();
+    } else {
+        res.status(401).send("Not an admin");
+    }
+};
+
+app.locals.POSSIBLE_STATES = POSSIBLE_STATES;
+app.locals.CURRENT_STATE = POSSIBLE_STATES.noVote;
+
+app.get("/api/me", isSignedIn, (req, res) => {
+    const user = req.user;
+
+    res.send({ ...user, voteIT: { admin: user.cid === app.get("adminCid") } });
+});
+
+app.post("/api/code", (req, res) => {
+    const { code } = req.body;
+
+    postGammaToken(app, code)
+        .then(response => {
+            const { access_token } = response.data;
+
+            const data = jwt.decode(access_token);
+            const user = _.find(that.gammaUsers, ["cid", data.user_name]);
+
+            if (user == null) {
+                res.status(401).send(
+                    "Please ask the counter to update the Gamma cache."
+                );
+                return;
+            }
+
+            res.cookie(AUTH_COOKIE_NAME, access_token, {
+                path: "/",
+                expires: new Date(Date.now() + 24 * 3600000), //24h
+                httpOnly: true,
+                secure: production,
+                sameSite: "strict",
+            });
+
+            const userId = user.id;
+
+            if (req.user.cid === app.get("adminCid")) {
```

Comment:
Keep getting `TypeError: Cannot read property 'cid' of undefined
    at /VoteIT/index.js:199:26` when logging in.

Still works, but checking if cid is defined would be nice.